src/controllers/StudentController.js

```diff
@@ -9,6 +9,18 @@ module.exports = {
       return res.status(500).send(error);
     }
   },
+  async getbyGroupId(req, res, next) {
+    const { groupId } = req.query;
+    if (groupId) {
+      try {
+        const students = await Student.findAll({ where: { groupId } });
+        return res.status(200).send(students);
+      } catch (error) {
+        return res.status(500).send(error);
+      }
+    }
```

Comment:
Добавить проверку по студентам, у которых нет групп:
```suggestion
    } else if (groupId === '') {
      const studentWoGroup = await Student.findAll({ where: { groupId: null } });
      return res.status(200).send(studentWoGroup);
    }
```
с фронта можно передавать dispatch(getStudentsByGroup('')